desc/plotting.py

```diff
@@ -728,17 +729,19 @@ def plot_fsa(
 
     fig, ax = _format_ax(ax, figsize=kwargs.get("figsize", (4, 4)))
 
-    values = np.array([])
-    for i, r in enumerate(rho):
-        if r > 0:
-            grid = LinearGrid(M=M, N=N, NFP=1, rho=np.array(r))
-            g, _ = _compute(eq, "sqrt(g)", grid)
-            data, label = _compute(eq, name, grid, kwargs.get("component", None))
-            values = np.append(values, np.mean(data * g) / np.mean(g))
-        elif r == 0:
-            grid = LinearGrid(M=0, N=0, NFP=1, rho=np.array(0.0))
-            data, label = _compute(eq, name, grid, kwargs.get("component", None))
-            values = np.append(values, np.mean(data))
+    grid = LinearGrid(M=M, N=N, NFP=1, rho=rho)
+    g, _ = _compute(eq, "sqrt(g)", grid, reshape=False)
+    data, label = _compute(eq, name, grid, kwargs.get("component", None), reshape=False)
+    values = compress(grid, surface_averages(grid, q=data, sqrt_g=g))
+
+    # TODO: Do we want to keep this logic?
+    # FIXME: Should I add it to surface_averages()?
+    #  I think currently surface_averages() could return nan at axis.
+    if 0 in rho:
+        grid = LinearGrid(M=0, N=0, NFP=1, rho=np.array(0.0))
+        data, label = _compute(eq, name, grid, kwargs.get("component", None))
+        values[0] = np.mean(data)
```

Comment:
`surface_averages` uses `jax` and will return an immutable array, so this line will error.

The easy way to fix is add `values = np.copy(values)` before line 743.

The cleaner way would be to use the method jax recommends. However, I think this functionality is deprecated. Could be wrong.
`TypeError: '<class 'jaxlib.xla_extension.DeviceArray'>' object does not support item assignment. JAX arrays are immutable. Instead of ``x[idx] = y``, use ``x = x.at[idx].set(y)`` or another .at[] method: https://jax.readthedocs.io/en/latest/jax.ops.html`

Reply:
I deleted this logic. Most of the compute data is undefined at the magnetic axis because the magnetic field is proportional to `psi_r / sqrt(g)` and `sqrt(g) = 0` at `rho = 0`. So the root cause of the problem is not the `surface_averages` function, and this logic wasn't helping anyways. 

Physically, all the compute quantities should be defined at the magnetic axis. But we would have to do l'Hospital's rule to compute them, which is more expensive than it's worth. So we avoid the problem by not having any grid nodes at rho=0